src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java

```diff
@@ -96,6 +104,72 @@ public void testProperFileExtension() throws Exception {
         verify(checkConfig, file.getPath(), expected1);
     }
 
+    /**
+     * This test is needed for 100% coverage.
+     * The Pitest reports some conditions as redundant, for example:
+     * <pre>
+     *     if (!collection.isEmpty()) { // This may me omitted.
+     *         Object value = doSomeHardJob();
+     *         for (Item item : collection) {
+     *             item.accept(value);
+     *         }
+     *     }
+     * </pre>
+     * But we really want to avoid calls to {@code doSomeHardJob} method.
+     * To make this condition mandatory, we need to broke one branch.
+     * In this case, mocking {@code TreeWalkerAuditEvent} will cause
+     * {@code getFilteredViolations} to fail. This prevents the condition
+     * <pre>
+     *     if (filters.isEmpty())
+     * </pre>
+     * in {@link TreeWalker#processFiltered(File, FileText)} to survive with Pitest mutations.
+     *
+     * @throws Exception if an error occurs
+     */
+    @Test
+    public void testNoAuditEventsWhenNoFilters() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(OneTopLevelClassCheck.class);
+        final String[] expected = {
+            "5:1: " + getCheckMessage(OneTopLevelClassCheck.class,
+                    OneTopLevelClassCheck.MSG_KEY, "InputTreeWalkerInner"),
+        };
+        try (MockedConstruction<TreeWalkerAuditEvent> mocked =
+                 Mockito.mockConstruction(TreeWalkerAuditEvent.class, (mock, context) -> {
+                     throw new CheckstyleException("No audit events expected");
+                 })) {
+            verify(checkConfig, getPath("InputTreeWalker.java"), expected);
+        }
+    }
+
+    @Test
+    public void testNoProcessWhenNoOrdinaryChecks() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(JavadocParagraphCheck.class);
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+        final DetailAST rootAst = mock(DetailAST.class);
+        // Ensure that there is no calls to walk(..., AstState.ORDINARY)
+        doThrow(IllegalStateException.class).when(rootAst).getFirstChild();
+        try (MockedStatic<JavaParser> parser = Mockito.mockStatic(JavaParser.class)) {
+            parser.when(() -> JavaParser.parse(any(FileContents.class))).thenReturn(rootAst);
+            // This will re-enable walk(..., AstState.WITH_COMMENTS)
+            parser.when(() -> JavaParser.appendHiddenCommentNodes(rootAst)).thenReturn(null);
```

Comment:
better to make violation in input, as no violations also means no check executions.

Reply:
done. Same for `testConditionRequiredWithoutCommentChecks`